sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueListener.cs

```diff
@@ -444,185 +451,14 @@ internal static void RegisterSharedWatcherWithQueueProcessor(QueueProcessor queu
             }
         }
 
-        public ScaleMonitorDescriptor Descriptor
-        {
-            get
-            {
-                return _scaleMonitorDescriptor;
-            }
-        }
-
-        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
-        {
-            return await GetMetricsAsync().ConfigureAwait(false);
-        }
-
-        public async Task<QueueTriggerMetrics> GetMetricsAsync()
-        {
-            int queueLength = 0;
-            TimeSpan queueTime = TimeSpan.Zero;
-
-            try
-            {
-                QueueProperties queueProperties = await _queue.GetPropertiesAsync().ConfigureAwait(false);
-                queueLength = queueProperties.ApproximateMessagesCount;
-
-                if (queueLength > 0)
-                {
-                    PeekedMessage message = (await _queue.PeekMessagesAsync(1).ConfigureAwait(false)).Value.FirstOrDefault();
-                    if (message != null)
-                    {
-                        if (message.InsertedOn.HasValue)
-                        {
-                            queueTime = DateTime.UtcNow.Subtract(message.InsertedOn.Value.DateTime);
-                        }
-                    }
-                    else
-                    {
-                        // ApproximateMessageCount often returns a stale value,
-                        // especially when the queue is empty.
-                        queueLength = 0;
-                    }
-                }
-            }
-            catch (RequestFailedException ex)
-            {
-                if (ex.IsNotFoundQueueNotFound() ||
-                    ex.IsConflictQueueBeingDeletedOrDisabled() ||
-                    ex.IsServerSideError())
-                {
-                    // ignore transient errors, and return default metrics
-                    // E.g. if the queue doesn't exist, we'll return a zero queue length
-                    // and scale in
-                    _logger.LogWarning($"Error querying for queue scale status: {ex.Message}");
-                }
-            }
-
-            return new QueueTriggerMetrics
-            {
-                QueueLength = queueLength,
-                QueueTime = queueTime,
-                Timestamp = DateTime.UtcNow
-            };
-        }
-
-        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
-        {
-            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.Cast<QueueTriggerMetrics>().ToArray());
-        }
-
-        public ScaleStatus GetScaleStatus(ScaleStatusContext<QueueTriggerMetrics> context)
+        public ITargetScaler GetTargetScaler()
         {
-            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
+            return _targetScaler != null ? _targetScaler.Value : null;
```

Comment:
`_targetScaler.Value` can not be null.

```suggestion
            return _targetScaler.Value;
```